### PR TITLE
API for creating command sender which forwards feedback

### PR DESCRIPTION
**Pull request by jpenilla**

### CODE REVIEW

**Review by kashike — APPROVED** [review]

File: patches/server/0861-API-for-creating-command-sender-which-forwards-feedb.patch [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+diff --git a/src/main/java/org/bukkit/craftbukkit/command/ServerCommandSender.java b/src/main/java/org/bukkit/craftbukkit/command/ServerCommandSender.java
+index eaff8df6c8c12c64e005a68a02e2e35ed88f757c..1c638a4b1f2c841928d8b2a7ae43e4ebb1f7eac7 100644
+--- a/src/main/java/org/bukkit/craftbukkit/command/ServerCommandSender.java
++++ b/src/main/java/org/bukkit/craftbukkit/command/ServerCommandSender.java
+@@ -13,9 +13,15 @@ import org.bukkit.plugin.Plugin;
+ 
+ public abstract class ServerCommandSender implements CommandSender {
+     private static PermissibleBase blockPermInst;
+-    private final PermissibleBase perm;
++    public final PermissibleBase perm; // Paper
```
Comment (Machine-Maker) [inline patches/server/0861-API-for-creating-command-sender-which-forwards-feedb.patch:152]: AT?

**Review by JRoy — APPROVED** [review]
troll face

**Comment by molor** [comment]
Hey, can you add second argument to `Consumer` that holds the command string so we will know to what is that `Component` related? Or if it possible to do different actions inside `Consumer` with `Component` based on command that passed to `Server#dispatchCommand`, let me know how, please

**Comment by electronicboy** [comment]
Consumer is a JDK class which only takes one arg, nor do we know what command is being executed when people call the send methods on that, so that's impossible. There is nothing stopping you passing the command you're running into a lambda or a class which extends Consumer and stores that info, etc

**Pull request merged**